Autopilot/PathManager/Src/PathManagerInterface.cpp

```diff
@@ -0,0 +1,8 @@
+#include "PathManagerInterface.h"
+#include "pathManager.hpp"
+
+static pathManager pathMng;
+
+void PathManagerInterfaceExecute(void) {
+    pathMng.execute();
```

Comment:
there's a caveat here. Calling execute once only executes a single state. But you want to run the full state machine once per FREERTOS thread execution. There's a mechanism to know wether the state machine is done executing or not. Take a look at how it's called in ZeroPilot-SW\Autopilot\Simulation\SimDriver\SimMain.cpp

Reply:
So looking at the solution in simulation that you guided me too, it seems as if they use a while loop. Is there a possible non-blocking solution to this or is this good enough for our purposes?

Reply:
Not sure what you mean by blocking/non blocking here. Usually, blocking means a thread busy waits without doing any meaningful work. THat's not what happens here. All the while loop ensures is that the state machine is called the appropriate number of times to execute 1 full cycle. (It should execute 1 full cycle per FREERTOS thread call)